agent/tool-scripts/base-tool

```diff
@@ -759,7 +759,7 @@ start)
 	# Force LANG=C for all tools
 	printf -- "#!/bin/bash\n# base-tool generated command file\n\nLANG=C PYTHONUNBUFFERED=True%s%s%s exec %s\n" "${_goroot}" "${_gopath}" "${_newpath}" "${tool_cmd}" > "${tool_cmd_file}"
 	chmod +x "${tool_cmd_file}"
-	printf -- "%s: running \"%s\"\n" "${tool_name}" "${tool_cmd}"
+	printf -- "%s: running \"%s\"\n" "${tool}" "${tool_cmd}"
```

Comment:
Hmm; you missed some gold files. At the same time, I wonder about the change in the prefix; do we really want to be identifying the "root" `${tool}`, which is generic, vs the full `${tool_name}` here?

```
*** 1,4 ****
! user-tool-foo42: running "/var/tmp/pbench-test-tool-scripts/user-tool/tmp/foo42.start /var/tmp/pbench-test-tool-scripts/user-tool/tools-group/user-tool-foo42 10"
! user-tool-foo42: stopping
! user-tool-foo42: post-processing following stop
! user-tool-foo42: post-processing data
--- 1,4 ----
! user-tool: running "/var/tmp/pbench-test-tool-scripts/user-tool/tmp/foo42.start /var/tmp/pbench-test-tool-scripts/user-tool/tools-group/user-tool-foo42 10"
! user-tool: stopping
! user-tool: post-processing following stop
! user-tool: post-processing data
```

Reply:
(Sorry about the missing gold files!)

>I wonder about the change in the prefix

While this is certainly open for discussion, this change was intentional:  the "tool" which the user invoked is `user-tool`, and so it seems like it should identify itself as such.  Now, it's true that `user-tool` is actually used to run another tool, so...it's not crazy to use the combined prefix, but it _is_ a little weird.  (In particular, it's not quite right in the first line of the output that you quoted, since that's describing `user-tool`'s invocation of the user's tool; and, if we go with the straightforward prefix for that one, then the others look strange, especially since they are describing actions by `user-tool` itself -- if the sub-tool reports something, it can tag it with it's own name....)

Now, if someone points out that a user could have a single Pbench run with multiple tools being run by separate invocations of `user-tool`, I might be swayed against this change, but I think the outputs go to separate log files, in which case any ambiguity created by the common prefix should be easy to dispel.

(I suppose, we might compromise by using a different format for the prefix (e.g. `user-tool(foo42): `) which would be clearer (i.e., there _is_ no `user-tool-foo42` command or script, but there _was_ an invocation of `user-tool` with the **argument** `foo42`).)

Reply:
Since `tool_name` is used to form the "tool output directory", and all remaining uses of `tool_name` after this change refer to files in the `tool_output_directory`, it seems we can just drop all other uses of it entirely.

Reply:
I'm not sure `user-tool` alone is actually of much use to anyone (it can't be used and has no meaning without `--tool-name`)... none of the other tools linked to this script have that problem.

I like the `user-tool(foo42)` idea, though, instead of implying a tool path that doesn't exist.

Reply:
> Since `tool_name` is used to form the "tool output directory", and all remaining uses of `tool_name` after this change refer to files in the `tool_output_directory`, it seems we can just drop all other uses of it entirely.

I'm not sure what you're suggesting, here, @portante....  With my change, the only references to `tool_name` are in the context of constructing various filenames; however, not all of them are (obviously) in `tool_output_directory` (so I cannot, e.g., simply replace uses of `tool_name` with `tool_output_directory`), so I think we still need to use `tool_name` in the places where I haven't replaced it with `tool`.  Or, are you saying something else?

Reply:
> I like the `user-tool(foo42)` idea, though, instead of implying a tool path that doesn't exist.

The principal problem there is that there are points in the script where we know we're running `user-tool` but don't yet know that we're using it to run `foo42`...so, if I go with the decorated approach, many of the messages will have the decoration but more than a quarter of them won't...which feels weird.